Fixed permissions issue with SGE job submission script

ericsuh opened the pull request:
The generated `run_miso.sh` file needed to have a `chmod +x`.

ericsuh committed:
Fixed permissions issue with SGE job submission script

yarden commented:
Hi Eric, thanks for this. I am traveling but will write a more detailed response when I return to the US. Did you need the executable when not running on the cluster? In all of our runs on either Linux or Mac OS X we've never needed the script to be executable, interested to know under what conditions you got this error? Best, Yarden

Sent from a mobile device

On Mar 21, 2012, at 5:16 AM, "Eric J. Suh"reply@reply.github.com wrote:

> The generated `run_miso.sh` file needed to have a `chmod +x`.
> 
> You can merge this Pull Request by running:
> 
>  git pull https://github.com/ericsuh/MISO fastmiso
> 
> Or you can view, comment on it, or merge it online at:
> 
>  https://github.com/yarden/MISO/pull/20
> 
> -- Commit Summary --
> - Fixed permissions issue with SGE job submission script
> 
> -- File Changes --
> 
> M misopy/cluster_utils.py (1)
> 
> -- Patch Links --
> 
>  https://github.com/yarden/MISO/pull/20.patch
>  https://github.com/yarden/MISO/pull/20.diff
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/yarden/MISO/pull/20

ericsuh commented:
Hi Yarden. I've actually only tried to run this on the SGE-driven cluster, but the given change fixed the problem. If it helps, I was using the following command line:

```
python run_events_analysis.py --compute-genes-psi hg19/TandemUTR.hg19\
elj-rnaseq/p-tophat.sorted.bam --output-dir elj-rnaseq/miso-out/p-tophat-output/\
--read-len=100 --use-cluster --SGEarray --job-name elj-miso-p
```

The fix is specific to the `run_SGEarray_cluster` function, as the other cluster function already seems to do this. Perhaps it has something to do with the permissions under which I am running the scripts?

-- Eric

ericsuh committed:
Added preflight options to SGE job script.

For situations in which the SGE job execution environment
needs to be tweaked (say, if you want to set PATH or PYTHONPATH)
before running MISO, added a `cluster_preflight` option in the
configuration file and altered the script to add that command
into the beginning of the job script.

ericsuh committed:
Fixed typo
